presto-main/src/main/java/com/facebook/presto/sql/analyzer/StatementAnalyzer.java

```diff
@@ -1218,6 +1217,10 @@ protected Scope visitTable(Table table, Optional<Scope> scope)
             Optional<ConnectorMaterializedViewDefinition> optionalMaterializedView = session.getRuntimeStats().profileNanos(
                     GET_MATERIALIZED_VIEW_TIME_NANOS,
                     () -> metadata.getMaterializedView(session, name));
+            // Prevent INSERT query when selecting from a materialized view. When `expandedQuery` is present, it is a REFRESH event.
+            if (optionalMaterializedView.isPresent() && "INSERT".equals(analysis.getUpdateType()) && !analysis.getExpandedQuery().isPresent()) {
```

Comment:
Hi @zacw7 , this condition looks hacky to me; with String comparison, the presence of expandedQuery, it's hard for new people who make change in the logic path to find this condition and get it correct.

Do you think it's better to take an allowlist approach and enumerate the allowed types of analysis.getStatement()? We'll allow RefreshMaterializedView, CreateMaterializedView, Query, DropMaterializedView, etc.

Reply:
That sounds a better approach to me. It's just I'm not sure if this `updateType` is used somewhere else so didn't modify it.